### PR TITLE
add tllist library

### PR DESCRIPTION
**Pull request by aacebedo**
Specify library name and version:  **tllist/1.0.5**

Not the author of the library but a project I am working on needs it.
---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by aacebedo** [comment]
Sadly I do not have access to logs :( Don't know what is happening.
Conan center hooks are okay locally

**Comment by jgsogo** [comment]
Hi! This is failing at the `conan export "recipes/tllist/all/conanfile.py" tllist/1.0.5@` command. Let me try it locally.

**Comment by jgsogo** [comment]
Oh! You are requiring Conan v1.44 and we are still using 1.43.1 here in CCI

```
ERROR: Error loading conanfile at '/home/conan/conan-center-index/recipes/tllist/all/conanfile.py': Current Conan version (1.43.1) does not satisfy the defined one (>=1.44.0).
```

We should propagate this error (at this step) to the user... we try to hide everything _unexpected_ just in case, but for sure this step can raise this way and the error needs to be in the output.

File: recipes/tllist/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import os
+
+from conans import ConanFile, Meson, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.44.0"
```
Comment (jgsogo) [inline recipes/tllist/all/conanfile.py:6]: Currently, we use `1.43.1`, so this PR won´t succeed until we change the version (usually not before one month after the release)

File: recipes/tllist/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("Only Linux supported")
+
+    def build_requirements(self):
+        self.tool_requires("pkgconf/[>=1.7.4]")
+        self.tool_requires("meson/[>=0.60.0]")
+        self.tool_requires("ninja/[>=1.10.0]")
```
Comment (jgsogo) [inline recipes/tllist/all/conanfile.py:32]: We don´t use version ranges for requirements... but for pure build-requires it is something we could consider if we start to use always two profiles (not-yet). wdyt @danimtb ?

File: recipes/tllist/all/conanfile.py (hunk 1 above)
Comment (aacebedo) [reply recipes/tllist/all/conanfile.py]: Ok I corrected this. However the instructions of the PR process explicitly requires and points to the latest release (ie 1.44.0). Maybe it shall be updated to add the one month period requirement ?

File: recipes/tllist/all/conanfile.py (hunk 2 above)
Comment (aacebedo) [reply recipes/tllist/all/conanfile.py:32]: > Hi! This is failing at the `conan export "recipes/tllist/all/conanfile.py" tllist/1.0.5@` command. Let me try it locally.

Thank you for the quick response !

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution. Please, consider https://github.com/aacebedo/conan-center-index/pull/1

File: recipes/tllist/all/test_package/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    generators = "pkg_config"
+    settings = "os", "compiler", "build_type", "arch"
+
+    def build_requirements(self):
+        self.build_requires("pkgconf/[>=1.7.4]")
+        self.build_requires("meson/[>=0.60.0]")
+        self.build_requires("ninja/[>=1.10.0]")
+
+    def build(self):
+        meson = Meson(self)
```
Comment (SSE4) [inline recipes/tllist/all/test_package/conanfile.py:15]: meson-based test package, something unusual, but still welcomed!

File: recipes/tllist/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+from conans import ConanFile, Meson, tools
+import os
+
+
+class TestPackageConan(ConanFile):
+    generators = "pkg_config"
+    settings = "os", "compiler", "build_type", "arch"
+
+    def build_requirements(self):
+        self.build_requires("pkgconf/[>=1.7.4]")
```
Comment (SSE4) [inline recipes/tllist/all/test_package/conanfile.py:10]: we don't use version ranges, but as it's a test_package, should be harmless?

File: recipes/tllist/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("Only Linux supported")
+
+    def build_requirements(self):
+        self.build_requires("pkgconf/[>=1.7.4]")
+        self.build_requires("meson/[>=0.60.0]")
+        self.build_requires("ninja/[>=1.10.0]")
+
```
Comment (SSE4) [inline recipes/tllist/all/conanfile.py:33]: please don't use version ranges here

**Review by SSE4 — CHANGES_REQUESTED** [review]
version ranges are not allowed in CCI
https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#why-are-version-ranges-not-allowed

**Pull request closed without merging**